qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py

```diff
@@ -0,0 +1,203 @@
+
+        if not calibrations:
+            calibrations = {}
+
+        # Add the parameters with their calibrated values to the table
+        for param_name, param in params.items():
+            self._add_parameter(param, calibrations.get(param_name, None))
+
+        gate = Gate(name=name, num_qubits=1, params=[_ for _ in params.values()])
+
+        if duration:
+            params['duration'] = duration
+
+        schedule = Schedule(name=name)
+        schedule = schedule.insert(0, Play(pulse_envelope(**params), channel))
+
+        if isinstance(channel, ControlChannel):
+            # TODO Need a mapping from ControlChannel to qubits
+            # TODO This works for CR, would it work for TC?
+            raise NotImplemented
+        else:
+            qubit = int(channel.name.replace('d', '').replace('m', ''))
+
+        circ = QuantumCircuit(1)
+        circ.append(gate, [0])
+        circ.add_calibration(gate, [qubit], schedule)
+
+        self._instructions[(name, (qubit, ))] = circ
+
+    def _basic_inst_parameters(self, name: str, channel: PulseChannel,
+                               pulse_envelope: ParametricPulse) -> Dict[str, ParameterExpression]:
+        """
+        This functions avoids the user to have to manage parameters and their name.
+
+        TODO Is there a better way of doing this?
+        """
+        params = {}
+        if pulse_envelope.__name__ in ['Drag', 'Gaussian', 'GaussianSquare']:
+            params['amp'] = Parameter(name + '.' + channel.name + '.amp')
+            params['sigma'] = Parameter(name + '.' + channel.name + '.sigma')
+
+        if pulse_envelope.__name__ == 'Drag':
+            params['beta'] = Parameter(name + '.' + channel.name + '.beta')
+
+        if pulse_envelope.__name__ == 'GaussianSquare':
+            params['width'] = Parameter(name + '.' + channel.name + '.width')
+
+        return params
+
+    def add_composite_instruction(self, inst_name: str, instructions: List[List[Tuple[str, set]]]):
+        """
+        Creates a new instruction from the given list of instructions.
+
+        Args:
+            inst_name: Name of the composite instruction to add
+            instructions: The instructions to add to the circuit. Instructions are grouped
+                by barriers to enforce relative timing between the instructions. Each instruction
+                is specified by its name and the qubit it applies to.
+        """
+        all_qubits = set()
+        for inst_list in instructions:
+            for inst_config in inst_list:
+                all_qubits |= set(inst_config[1])
+
+        circ = QuantumCircuit(max(all_qubits)+1)
+        for inst_list in instructions:
+            for inst_config in inst_list:
+                name = inst_config[0]
+                qubits = inst_config[1]
+                qc = self.get_instruction_template(name, qubits)
+                circ.compose(qc, qubits=qubits, inplace=True)
```

Comment:
Please make sure that compose does substitution of calibration rather than copy. If it does copy, you cannot refer the original calibration from composed circuit. You can check this with `id` function.

Reply:
Does this https://github.com/Qiskit/qiskit-terra/pull/5396 workout?

Reply:
This is not needed anymore.